### PR TITLE
Create and add Table of Contents to about page

### LINKED ISSUE #200: Create floating index for About page
Our [About page][about] is growing and is likely to grow further. As it grows, it's becoming a little bit confusing to keep track of where you are on the page, and it's not currently possible to see an overview. We should create something like a floating table of contents in a sidebar so that users can keep track of the page's contents. On mobile, the table of contents could move to the top, perhaps collapsed by default.

[about]: https://opensourcepledge.com/about

### PR DESCRIPTION
**Pull request by safwanyp**
Closes #200 

---

I did not want to touch any of the existing code outside the `about` page, which is why one might notice some repeated styles. I did this to make sure that even if this PR is merged, nothing other than the about page is affected. 

I ran `pnpm audit --fix` to address the vulnerabilities, but if it seems unnecessary I can get rid of that commit. 

I noticed guidelines for [Copyright Headers](https://github.com/opensourcepledge/opensourcepledge.com/blob/862c9b63c7ab16edea29fb64e8eca22b9ec1274f/CONTRIBUTING.md#copyright-headers) in the contributing guidelines but wasn't sure if I should add anything. If I need to, please do let me know 🙏🏻

Feedback is appreciated!

### CODE REVIEW

**Comment by vladh** [comment]
Thank you @safwanyp! I'll review this soon. Just a quick note: you should definitely add copyright headers.

**Comment by chadwhitacre** [comment]
I think this is a net positive, thanks @safwanyp. I would make it always shown (no collapse) and no border on it, and probably an h2 or h3 for the "Table of Contents" headers. It looks like the anchors are manual. Maybe a follow-up to this (separate PR) would be to make them dynamic.

image

**Comment by safwanyp** [comment]
Noted! How does this look? Will push a commit if it looks good 😄 
osp-sc

**Comment by chadwhitacre** [comment]
Left looks good. What is the right screenshot showing?

**Comment by safwanyp** [comment]
> Left looks good. What is the right screenshot showing?

Right is the desktop (big screen) view. Do you want the big screen to not have the changes you mentioned?

**Comment by chadwhitacre** [comment]
Ah, seeing it now. Looks good. 👍

**Comment by safwanyp** [comment]
- made the change suggested by @chadwhitacre 
- rebased on origin/main to resolve conflicts without an extra commit
- force-pushed due to rebase

**Comment by chadwhitacre** [comment]
Hrm ... are the pnpm lockfile and npm upgrades strictly necessary?

**Comment by safwanyp** [comment]
> Hrm ... are the pnpm lockfile and npm upgrades strictly necessary?

Those were strictly due to me running `pnpm audit --fix` (it's part of my install alias 😅). If you prefer, I can just remove that commit completely

**Comment by chadwhitacre** [comment]
Yes please, `+4,820 −567` is a scary diff. :)

**Comment by safwanyp** [comment]
Yeah i think I made it worse by using `pnpm` instead of `npm` 😂 
Anyways, I've removed that commit completely, and force pushed.

**Comment by chadwhitacre** [comment]
Thanks ... now seeing this detail: the blob is behind the TOC on wide screen, kinda unideal for it to get cutoff, but presumably it doesn't look great if the TOC background is transparent. 🤔

image

**Comment by chadwhitacre** [comment]
🤔🤔🤔🤔🤔

image

**Comment by safwanyp** [comment]
Yes the issue was that background being transparent made the contents hard to read, when over a blob. This is my opinion, but if you think it looks good, I can quickly change the styling!

**Comment by chadwhitacre** [comment]
Also this is more of an edge case but at wide width and short height the TOC is cutoff with no way to access lower items.

image

I'm leaning towards using the narrow-width layout universally. Wdyt?

**Comment by chadwhitacre** [comment]
> If I understand correctly, you're leaning towards having the TOCs at the top always?

Yes. Seem reasonable to get this done w/o getting bogged down in design details?

> I'll be away for the rest of the night (CET).

👍 Thanks for tackling this.

**Comment by safwanyp** [comment]
> > If I understand correctly, you're leaning towards having the TOCs at the top always?
> 
> Yes. Seem reasonable to get this done w/o getting bogged down in design details?

Yeah that sounds completely valid and fair. I'll push a commit soon

**Comment by safwanyp** [comment]
Hey @chadwhitacre @vladh 
I've pushed all the latest changes. Please have a look and let me know what you think!

Brief summary of latest changes:
- I've moved the table of contents to the top of the page (under the page heading) across all views, to make sure it's scoped to the actual content.
- I've removed all the duplicated classes as they were only needed to make sure the "floating" worked.

File: src/pages/about.astro [hunk 1]
```diff
@@ -25,254 +27,256 @@ const members = filterInactiveMembers(await getMembers());
...
+          <p
+            >This is why our main goal is <strong>to get companies to pay maintainers</strong> of software they depend on.
+            We believe that by doing this, we can contribute to a healthy, thriving and secure Open Source ecosystem. We
+            recognise this is a multifaceted problem, but the solution we are focusing on is providing <strong>direct
+            payments to maintainers</strong>, in order to support them and their work.</p
+          >
+        </section>
+
+        <section>
+          <h2>What does joining the Pledge entail?</h2>
```
Comment (chadwhitacre) [inline src/pages/about.astro:86]: Is there a reason not all FAQs are included in the TOC?

Comment (safwanyp) [reply src/pages/about.astro:86]: Ah shoot I think i messed up when resolving merge conflicts. will push a fix for this tomorrow!

Comment (safwanyp) [reply src/pages/about.astro:86]: Pushed a commit for this!

**Comment by safwanyp** [comment]
@vladh Appreciate the comments, will definitely help in the future 😁

**Comment by chadwhitacre** [comment]
Thanks for the contribution, @safwanyp!

**Pull request merged**